### DISCUSSION TITLE
Feiv does not calculate second stage R2 and F-statistic

### DISCUSSION THREAD

**Discussion by chungkhy** [py-econometrics/pyfixest#738] [Q&A]
Hi pyfixest community! Love the performance and convenience this package brings.

I have a question regarding Feiv. I can't seem to find where to get the second stage R2 and _F_stat. After looking through the files and Issues page, I could only find a description on R2 - _`feols()` does not report R2 statistics for IVs_ - and nothing about second stage F statistic. Is there a reason why R2 is not reported for IVs? Also, is it possible to get the second stage F statistic in an Feiv instance?

Thank you so much!

**Comment by s3alfisc** [comment]
Hi @chungkhy , you get the R2 value for the **first stage** but not for the "full IV" estimator. We have this in the docs [here ](https://py-econometrics.github.io/pyfixest/quickstart.html#instrumental-variables-iv-estimation)but maybe it is a little to hidden? I wonder if we should add a dedicated notebook on IV estimaton? 

The IV attributes are listed [here](https://py-econometrics.github.io/pyfixest/reference/estimation.feiv_.Feiv.html). After running an IV, you should get the first stage f-statistic as the `Feiv._f_stat_1st_stage` and the effective F statistic (if you have computed it) via `Feiv._eff_F`. 

Hope this helps!

**Comment by chungkhy** [comment]
Hi @s3alfisc, thanks for this 🙌

Yes, I am able to find all first stage diagnostics/statistics with no problem. My question is more specifically for the second stage or "full IV" regression results. There is no R2 value or F-stat for it at the moment to my understanding. Would it be possible to add these in future iterations of Feiv?
